cgen: can't print the function address.

### Describe the bug

```v
fn foo() {
}

fn main() {
	println(&foo)
}
```

### Expected Behavior

Successful compilation.

### Current Behavior

```
==================
   ^~~~~~~~~~~~~~~
/tmp/v_501/t.15329770610299642670.tmp.c:12227:25: error: expected expression
        println(main__foo_str(*));
                               ^
1 warning and 1 error generated.
```

### Reproduction Steps

Run code from `Describe the bug`.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.3.3 2d3508c

### Environment details (OS name and version, etc.)

macOS 13.0

--- a/vlib/v/gen/c/str.v
+++ b/vlib/v/gen/c/str.v
@@ -139,12 +139,12 @@ fn (mut g Gen) gen_expr_to_string(expr ast.Expr, etype ast.Type) {
 		is_var_mut := expr.is_auto_deref_var()
 		str_fn_name := g.get_str_fn(typ)
 		g.write('${str_fn_name}(')
-		if str_method_expects_ptr && !is_ptr {
-			g.write('&')
-		} else if (!str_method_expects_ptr && is_ptr && !is_shared) || is_var_mut {
-			g.write('*')
-		}
 		if sym.kind != .function {
+			if str_method_expects_ptr && !is_ptr {
+				g.write('&')
+			} else if (!str_method_expects_ptr && is_ptr && !is_shared) || is_var_mut {
+				g.write('*')
+			}
 			g.expr_with_cast(expr, typ, typ)
 		}
 		g.write(')')

--- a/vlib/v/tests/print_reference_function_test.v
+++ b/vlib/v/tests/print_reference_function_test.v
@@ -1,0 +1,7 @@
+fn foo() {
+}
+
+fn test_print_reference_function() {
+	println(&foo)
+	assert '${&foo}' == 'fn ()'
+}

